### PR TITLE
Change GridMap navigation_layers to per mesh_library item

### PR DESCRIPTION
**Pull request by smix8**
Changes GridMap `navigation_layers` from single value for the entire GridMap to a bitmask for each item used in the `mesh_library` with a baked `navmesh`.

Tbh a single value for the entire GridMap made little practical sense as there can only be ever a single navmesh in the same place for the same navigation map. To support different agent's that should be on the same navigation map while using a GridMap there must be an option to control the usage on an item / cell level.

### CODE REVIEW

File: doc/classes/MeshLibrary.xml [hunk 1]
```diff
@@ -59,6 +59,13 @@
 				Returns the item's name.
 			</description>
 		</method>
+		<method name="get_item_navigation_layers" qualifiers="const">
+			<return type="int" />
+			<param index="0" name="id" type="int" />
+			<description>
+				Returns the item's [code]navigation_layers[/code] bitmask.
```
Comment (akien-mga) [inline doc/classes/MeshLibrary.xml:66]: `[code]` markup with a snake_case identifier only makes sense if we're referring to actual code - but I don't see any `navigation_layers` property here, so this is confusing.

If this is an internal C++ member variable which is accessed via those methods, it doesn't make sense to refer to its C++ name which users don't know about. Instead, this should be described in plain English.

File: doc/classes/MeshLibrary.xml [hunk 2]
```diff
@@ -126,6 +133,14 @@
 				This name is shown in the editor. It can also be used to look up the item later using [method find_item_by_name].
 			</description>
 		</method>
+		<method name="set_item_navigation_layers">
+			<return type="void" />
+			<param index="0" name="id" type="int" />
+			<param index="1" name="navigation_layers" type="int" />
+			<description>
+				Sets the item's [code]navigation_layers[/code] bitmask.
```
Comment (akien-mga) [inline doc/classes/MeshLibrary.xml:141]: Same issue here.

File: modules/gridmap/doc_classes/GridMap.xml [hunk 3]
```diff
@@ -197,7 +182,7 @@
 	</methods>
 	<members>
 		<member name="bake_navigation" type="bool" setter="set_bake_navigation" getter="is_baking_navigation" default="false">
-			If [code]true[/code], this GridMap bakes a navigation region.
+			If [code]true[/code], this GridMap creates a navigation region for each cell that uses a mesh_library item with a navmesh. The create navigation region will use the [code]navigation_layers[/code] bitmask assigned to the mesh_library item.
```
Comment (akien-mga) [inline modules/gridmap/doc_classes/GridMap.xml:185]:
```suggestion
			If [code]true[/code], this GridMap creates a navigation region for each cell that uses a [member mesh_library] item with a navmesh. The created navigation region will use the [MeshLibrary]'s navigation layers bitmask defined using [method MeshLibrary.set_item_navigation_layers].
```
Tentative change to take into account what I described for the MeshLibrary doc change. I don't know if that's actually accurate.

File: doc/classes/MeshLibrary.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/MeshLibrary.xml:66]: Reading the code a bit more, I see it's a property defined dynamically for each Mesh item... not sure how best to describe this.

File: modules/gridmap/doc_classes/GridMap.xml (hunk 3 above)
Comment (smix8) [reply modules/gridmap/doc_classes/GridMap.xml]: Preferred solution would be to just remove the code tag and use a plain "navigation layers" text.

Not sure the MeshLibrary set_item_navigation_layers() should be mentioned here as this is usually something users set with the inspector gui for the mesh library and not by script.

File: doc/classes/MeshLibrary.xml (hunk 1 above)
Comment (smix8) [reply doc/classes/MeshLibrary.xml]: I think just using plain text would work best here,
[code]navigation_layers[/code] -> navigation layers

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
